pkg/controllers/federate/util.go

```diff
@@ -215,38 +198,68 @@ var (
 		util.OrphanManagedResourcesInternalAnnotation,
 		common.EnableFollowerSchedulingAnnotation,
 	)
+
+	federatedLabels = sets.New(
+		scheduler.PropagationPolicyNameLabel,
+		scheduler.ClusterPropagationPolicyNameLabel,
+		override.OverridePolicyNameLabel,
+		override.ClusterOverridePolicyNameLabel,
+	)
 )
 
+func classifyStringMap(
+	src map[string]string,
+	matcher func(key string) (federated, template bool),
+) (federatedMap, templateMap map[string]string) {
+	federatedMap = make(map[string]string, len(src))
+	templateMap = make(map[string]string, len(src))
+
+	for key, value := range src {
+		federated, template := matcher(key)
+		if federated {
+			federatedMap[key] = value
+		}
+		if template {
+			templateMap[key] = value
+		}
+	}
+
+	return federatedMap, templateMap
+}
+
 // Splits annotations from a source object into federated annotations and template annotations.
 func classifyAnnotations(annotations map[string]string) (
-	federated map[string]string,
-	template map[string]string,
+	federatedAnnotations map[string]string,
+	templateAnnotations map[string]string,
 ) {
-	result := map[annotationClass]map[string]string{}
-
-	for key, value := range annotations {
-		class := classifyAnnotation(key)
+	federatedAnnotations, templateAnnotations = classifyStringMap(annotations, classifyAnnotation)
+	federatedAnnotations[common.FederatedObjectAnnotation] = "1"
```

Comment:
what's the meaning of "1"?

Reply:
It means "true". This annotation is used to distinguish whether a federated object is created from a source object, or created manually. I agree it's not the best mechanism, but we probably shouldn't change it in this PR.

Reply:
> It means "true". This annotation is used to distinguish whether a federated object is created from a source object, or created manually. I agree it's not the best mechanism, but we probably shouldn't change it in this PR.

Got it